src/performance/manager.py

```
@@ -0,0 +1,182 @@
+        if operation not in self.active_operations:
+            raise ValueError(f"Operation {operation} is not being monitored")
+            
+        # Get metrics
+        metrics = self.monitor.end_measurement(operation)
+        self.active_operations.remove(operation)
+        
+        # Check if optimization is needed
+        should_optimize = self._should_optimize()
+        if should_optimize:
+            optimization_result = self.optimizer.optimize_operation(operation, metrics)
+            self.last_optimization = datetime.now()
```

Ruff (0.8.2)

`datetime.datetime.now()` called without a `tz` argument

(DTZ005)